source/Octopus.Tentacle.Tests.Integration/Util/TcpTentacleHelpers/PollingResponseMessageTcpKiller.cs

```diff
@@ -50,9 +50,23 @@ public void PauseConnectionOnNextResponse(Action? callBack = null)
 
         public IDataTransferObserver DataTransferObserver()
         {
+            int numberOfWritesFromTentacleSeen = 0;
             return new DataTransferObserverBuilder().WithWritingDataObserver((tcpPump, dataFromTentacle) =>
             {
-                var size = dataFromTentacle.Length;
+                numberOfWritesFromTentacleSeen++;
+
+                if (pauseConnection || killConnection)
+                {
+                    // For polling tentacle to connect it first sends some ssl stuff, then a MX control message to
+                    // register itself and finally a control PROCEED message. This means it is not until the 4th
+                    // write that we could be processing a message.
+                    if (numberOfWritesFromTentacleSeen <= 4)
```

Comment:
I don't even want to know how/why you had to figure this out 🙈

Reply:
It is a bit of a pain, and I am honestly not super happy. But I can't think of a better way unless we either just throw fake network exceptions OR we introduce a halibut level proxy to man in the middle the test.

Reply:
I think the comment to explain the magic number is good enough for our purposes.